### PR TITLE
Scripts: add perf.sh to do performance profiling

### PR DESCRIPTION
**Pull request by changsh726**

### CODE REVIEW

File: scripts/perf.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+function main() {
+  if [ "$#" -eq 0 ]; then
+    print_usage
+    exit 1
+  fi
+
+  local opt="$1"
+  shift
+  case "${opt}" in
```
Comment (storypku) [inline scripts/perf.sh:115]: Add a "clean" option to cleanup  outputs

File: scripts/perf.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    rm -f "${OUTPUT_DIR}${PERF_DATA}"
+  fi
+}
+
+# Note(All): if permission denied, run `echo 1 > /proc/sys/kernel/perf_event_paranoid` as root user out of docker
+function gen_perf_data() {
+  _check_perf
+  local module=$1
+  _check_out_dir
+  ${PERF_CMD} record -e cpu-clock -o "${OUTPUT_DIR}${PERF_DATA}" -g mainboard -d \
```
Comment (lx18233184051) [inline scripts/perf.sh:73]: what does mainboard mean?

Comment (changsh726) [reply scripts/perf.sh:75]: use mainboard to start the module

Comment (lx18233184051) [reply scripts/perf.sh]: mainboard is a process ?

Comment (changsh726) [reply scripts/perf.sh]: yep

**Pull request merged**